File: src/main/java/it/contrader/dto/StoryDTO.java
package it.contrader.dto;

import java.util.List;

import com.fasterxml.jackson.annotation.JsonIdentityInfo;
import com.fasterxml.jackson.annotation.ObjectIdGenerators;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
@JsonIdentityInfo(generator=ObjectIdGenerators.PropertyGenerator.class, property="idStory", scope= StoryDTO.class)
public class StoryDTO {
	
	private long idStory;
	
	private String title;
	
	private String plot;
	
	private boolean published;
	
	private UserDTO user;
	
	private CategoryDTO category;
	
	private NovelDTO novel;
	
//	private List<UserDTO> like;
	
	
}

File: src/dto/storydto.ts
import { CategoryDTO } from './categorydto';
import { UserDTO } from './userdto';
import { NovelDTO } from './novelsdto';

export class StoryDTO {
    
    idStory: number;

    title: string = '';

    plot: string = '';

    published: boolean = false;

    category: CategoryDTO = new CategoryDTO();

    user: UserDTO = null;

    // like: UserDTO[] = null;

    // bookmark: UserDTO[] = null;

    novel: NovelDTO = null;
}
